### PR TITLE
REFACTOR: chat-message-info

### PR DESCRIPTION
**Pull request by jjaffeux**
Not this commit will also allow primary group flair to show next to username.

- [ ] chat-scoped -- core unaffected

### View mode

- [ ] docked (windowed/drawer)
- [ ] isolated (full-screen)

### Browsers

- [ ] safari
- [ ] chrome
- [ ] firefox

### Device

- [ ] desktop – wide screen
- [ ] mobile – `?mobile_view=1`

### Ember

- [ ] ember-cli
- [ ] legacy

### CODE REVIEW

**Comment by jjaffeux** [comment]
Screenshot 2022-05-13 at 00 10 34

File: assets/javascripts/discourse/templates/components/chat-message-info.hbs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        {{user-avatar-flair user=@message.user}}
+      </span>
+    </span>
+  {{/if}}
+
+  <span class="chat-message-info__date">
+    {{format-chat-date @message @details}}
+  </span>
+
+  {{#if (or @message.reviewable_id (eq @message.user_flag_status 0))}}
```
Comment (martin-brennan) [inline assets/javascripts/discourse/templates/components/chat-message-info.hbs:30]: I find this a little hard to follow, is it `if the message has a reviewable id AND the user flag status equals 0` or is it `if the message has a reviewable id OR the user flag status equals 0`?

**Review by martin-brennan — APPROVED** [review]
Looking really nice, love the new separation, makes chat-message a lot leaner :+1:

File: app/serializers/chat_message_serializer.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@ class ChatMessageSerializer < ApplicationSerializer
     :edited,
     :reactions
 
-  has_one :user, serializer: BasicUserSerializer, embed: :objects
+  has_one :user, serializer: UserCardSerializer, embed: :objects
```
Comment (davidtaylorhq) [inline app/serializers/chat_message_serializer.rb:16]: The UserCardSerializer includes a data from a *lot* of tables, so will be much slower. Do we need all of it?

To avoid N+1s, we'll need to make sure all the data is preloaded like

https://github.com/discourse/discourse/blob/b65ecf69871656155c381343cd84de919cbbf803/app/controllers/users_controller.rb#L112-L120

Comment (jjaffeux) [reply app/serializers/chat_message_serializer.rb:16]: Yep this is why I pointed to it in code-review. No we don't need all of it, we just need the flair stuff. But In the end that would mean a custom user serializer no?

Comment (davidtaylorhq) [reply app/serializers/chat_message_serializer.rb:16]: Yeah either a custom user serializer, or you could wrap up the user in a ChatPosterSerializer (similar to core's TopicPosterSerializer)

https://github.com/discourse/discourse/blob/e599b5b08f4323d50fb2dd584dbb06333d60b995/app/serializers/topic_poster_serializer.rb

Comment (jjaffeux) [reply app/serializers/chat_message_serializer.rb:16]: Will look at this, was looking for something like this, thanks
